test/net_http_test.rb

```diff
@@ -511,42 +492,15 @@ def close_circuit!(hostname: SemianConfig['toxiproxy_upstream_host'], toxic_port
 
   def with_server(ports: [SemianConfig['http_port']], reset_semian_state: true)
     ports.each do |port|
-      hostname = '0.0.0.0'
-      begin
-        server = nil
-        server_threw_error = false
-        server_thread = Thread.new do
-          Thin::Logging.silent = true
-          server = Thin::Server.new(hostname, port, RackServer)
-          begin
-            server.start
-          rescue StandardError
-            server_threw_error = true
-            raise
-          end
-        end
-
-        begin
-          poll_until_ready(hostname: hostname, port: port)
-        rescue RuntimeError
-          server_thread.kill
-          server_thread.join if server_threw_error
-          raise
-        end
-
-        assert(server.running?)
-        reset_semian_resource(hostname: SemianConfig['toxiproxy_upstream_host'], port: port) if reset_semian_state
+      if port == SemianConfig['http_port'] || SemianConfig['http_port'] + 100 # rubocop:disable all
+        reset_semian_resource(port: port) if reset_semian_state
         @proxy = Toxiproxy[:semian_test_net_http]
-        yield(hostname, port.to_i)
-      ensure
-        server&.stop
-        server_thread.kill
-        poll_until_gone(hostname: hostname, port: port)
+        yield('0.0.0.0', port.to_i)
```

Comment:
This is technically an implicit coupling to the `0.0.0.0` value used in `TestServer.start`. I'd recommend calling this out by putting it in a constant.

Arguably, though, it's not super useful to yield. `0.0.0.0` basically says that any local address that can route to this process will be accepted. I would think the more useful thing to yield would be an address that can be used to talk to the server (e.g., `127.0.0.1` in this case).

Reply:
I'll let @paranoidaditya comment on this, but in my experience with Docker, `0.0.0.0` is generally preferred to `127.0.0.1`.
But I agree that it should be moved into a named constant.

Reply:
For binding, or making a request?

I'll clarify my comment too: if the intent is to actually yield the address we're bound to, sticking with `0.0.0.0` is a-o-k. If we'd like to yield an address that routes to the server referenced by `with_server`, `0.0.0.0` would go through the "default" route, which I doubt we can guarantee would route to the server (but it probably generally does).

Reply:
Changed it get the value from the yaml, this was copied from previous code